### PR TITLE
General fixes.

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

File: osu.Framework/Graphics/Containers/Container.cs [hunk 1]
```diff
@@ -161,7 +161,7 @@ protected override void ApplyDrawNode(DrawNode node)
         public virtual IEnumerable<T> Children
         {
             get {
-                return Content != this ? Content.Children : children;
+                return Content != this ? Content?.Children : children;
             }
```
Comment (SirCmpwn) [inline osu.Framework/Graphics/Containers/Container.cs]: This seems like it's just hiding programmer errors.

Comment (peppy) [inline osu.Framework/Graphics/Containers/Container.cs]: i'll try and figure where i got this and see if it's feasible to fix without the null check.

File: osu.Framework/Input/InputManager.cs [hunk 2]
```diff
@@ -339,7 +341,7 @@ protected virtual void UpdateMouseState(InputState state)
                 pos.X /= currentCursorHandler.Size.X;
                 pos.Y /= currentCursorHandler.Size.Y;
 
-                Quad q = ScreenSpaceDrawQuad;
```
Comment (Tom94) [inline osu.Framework/Input/InputManager.cs]: I don't think Quad should be used here at all. You want to transform `pos` into the local coordinates of the inputmanager (which hopefully go from 0 to the resolution of the window, or alternatively whatever you set DrawSize to). You could then also apply some custom factor to scale things if you need to.

Comment (peppy) [inline osu.Framework/Input/InputManager.cs]: InputManagers can be placed at any point in the tree. In this case, a Player has an InputManager. Now consider that the Player may be inside a TestBrowser, which takes up some horizontal space.

This change allowed for

`protected override Quad InputManagerMappingQuad => Host.ScreenSpaceDrawQuad;`

to resolve this.

I'm open to better solutions.

**Pull request merged**